packages/xo-web/src/common/action-button.js

```diff
@@ -92,7 +94,20 @@ export default class ActionButton extends Component {
 
       const { redirectOnSuccess } = props
       if (redirectOnSuccess !== undefined) {
-        const to = typeof redirectOnSuccess === 'function' ? redirectOnSuccess(result, handlerParam) : redirectOnSuccess
+        if (redirectOnSuccess === ActionButton.GO_BACK) {
+          return this.context.router.goBack()
+        }
+        let to
+        switch (typeof redirectOnSuccess) {
+          case 'string':
+            to = redirectOnSuccess
+            break
+          case 'function':
+            to = redirectOnSuccess(result, handlerParam)
+            break
+          default:
+            throw new Error(`Redirect on success ${redirectOnSuccess} type unknown ${typeof redirectOnSuccess}`)
+        }
```

Comment:
Both of your commits modify this part, is that intended?

Reply:
I did a lot of commit rewriting for this fix. Since there is only two commit, maybe it would be better to squash them on merge

Reply:
Ok!